vignettes/articles/cheatsheet.Rmd

```diff
@@ -0,0 +1,557 @@
+---
+title: "Relational data with dm: Cheat sheet"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{cheatsheet}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+resource_files:
+  - cheatsheet
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  echo = FALSE
+)
+```
+
+[For printing](cheatsheet/cheatsheet-printable.html)
+
+<div class="page1">
+> If you have more than one table, use dm.
+
+<div class="page1col1">
+## <i class="fa-solid fa-circle-plus"></i> Create dm...
+
+### <i class="fa-solid fa-database"></i> from database: `dm_from_con()`
+
+```{r}
+get_src <- function(src) {
+  if (!nzchar(Sys.getenv("CHEATSHEET"))) {
+    file.path("cheatsheet-figures", src)
+  } else {
+    sprintf("%s/%s", "cheatsheet-figures", src)
+  }
+}
+get_svg <- function(path, alt) {
+  svg <- xml2::read_xml(path)
+  xml2::xml_remove(xml2::xml_find_all(svg, "//comment()"))
+    xml2::xml_attr(svg, "class") <- "cheatsheet-svg"
+    xml2::xml_attr(svg, "aria-labelledby") <- sprintf("svgDescription-%s", basename(path))
+    xml2::xml_add_child(
+      svg,
+      .value = "description",
+      alt,
+      .where = 0
+    )
+    # to drop the XML declaration that messes things up
+    temp_file <- withr::local_tempfile()
+    xml2::write_xml(svg, file = temp_file, options = "format_whitespace")
+    lines <- readLines(temp_file)
+    lines <- lines[!grepl("^<\\?", lines)]
+    lines <- lines[!grepl("^<!", lines)]
+    glue::glue_collapse(lines, sep = "")
+}
+create_figure <- function(class, src, alt, caption) {
```

Comment:
figures are handled differently for the vignette and for the printable cheatsheet. I can't remember why we inlined the SVGs for the vignette, I think it was to make less requests? Anyway both versions include the alternative text.